webrtc.html

```diff
@@ -625,6 +625,313 @@
           </li>
         </ol>
 
+        <p>To <dfn id=
+        "update-ice-gathering-state">update the ice gathering state</dfn> of an
+        <code><a>RTCPeerConnection</a></code> instance <var>connection</var> to
+        <var>newState</var>, the User Agent MUST queue a task that sets
+        <var>connection</var>'s <a href=
+        "#dom-peerconnection-ice-gathering-state">iceGatheringState</a>
+        attribute to <var>newState</var> and fires a simple event named
+        <code><a href=
+        "#event-icegatheringstatechange">icegatheringstatechange</a></code> at
+        <var>connection</var>.</p>
+
+        <p>To <dfn id=
+        "update-ice-connection-state">update the ice connection state</dfn> of an
+        <code><a>RTCPeerConnection</a></code> instance <var>connection</var> to
+        <var>newState</var>, the User Agent MUST queue a task that sets
+        <var>connection</var>'s <a href=
+        "#dom-peerconnection-ice-connection-state">iceConnectionState</a>
+        attribute to <var>newState</var> and fires a simple event named
+        <code><a href=
+        "#event-iceconnectionstatechange">icegatheringstatechange</a></code> at
+        <var>connection</var>.</p>
+
+        <p>To <dfn id="set-description">set an RTCSessionDescription</dfn>
+        <var>description</var> on an <code><a>RTCPeerConnection</a></code>
+        object <var>connection</var>, run the following steps:</p>
+
+        <ul>
+          <li>
+            <p>When the setter method is invoked, the user agent MUST run the
+            following steps:</p>
+
+            <ol>
+              <li>
+                <p>If <var>connection</var>'s
+                <a href="#dom-peerconnection-signaling-state">signaling
+                state</a> is <code>closed</code>, the user agent MUST return a
+                promise rejected with an <code>InvalidStateError</code>.</p>
+              </li>
+
+              <li>
+                <p>If <var>description</var> is set as a local description and
+                contains a different set of ICE credentials, then the ICE Agent
+                MUST trigger an ICE restart, and run the following steps:</p>
+
+                <ol>
+                  <li>
+                    <p>If <var>connection</var> is not already gathering,
+                    <a href="#update-ice-gathering-state">update
+                    <var>connection</var>'s ice gathering state</a> to
+                    <code>gathering</code>.</p>
```

Comment:
This doesn't match with the changes to add `RTCIceTransport`, since the gathering state will be deprecated (?) or turned into a composite state.  I think that Peter's rules for composite states need to be followed.